src/Domain/CoordinateMaps/ProductMapsTimeDep.tpp

```diff
@@ -0,0 +1,413 @@
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  using UnwrappedT = tt::remove_cvref_wrap_t<T>;
+  return product_detail::apply_jac(
+      source_coords, map1_, map2_,
+      [&time, &functions_of_time](const auto& point, const auto& map) noexcept {
+        return CoordinateMap_detail::apply_inverse_jacobian(
+            map, point, time, functions_of_time,
+            domain::is_jacobian_time_dependent_t<
+                std::decay_t<decltype(map)>,
+                std::reference_wrapper<const UnwrappedT>>{});
+      },
+      std::make_index_sequence<Map1::dim>{},
+      std::make_index_sequence<Map2::dim>{});
+}
+
+template <typename Map1, typename Map2>
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, ProductOf2Maps<Map1, Map2>::dim,
+         Frame::NoFrame>
+ProductOf2Maps<Map1, Map2>::jacobian(
+    const std::array<T, dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  using UnwrappedT = tt::remove_cvref_wrap_t<T>;
+  return product_detail::apply_jac(
+      source_coords, map1_, map2_,
+      [&time, &functions_of_time](const auto& point, const auto& map) noexcept {
+        return CoordinateMap_detail::apply_jacobian(
+            map, point, time, functions_of_time,
+            domain::is_jacobian_time_dependent_t<
+                std::decay_t<decltype(map)>,
+                std::reference_wrapper<const UnwrappedT>>{});
+      },
+      std::make_index_sequence<Map1::dim>{},
+      std::make_index_sequence<Map2::dim>{});
+}
+
+template <typename Map1, typename Map2>
+void ProductOf2Maps<Map1, Map2>::pup(PUP::er& p) {
+  p | map1_;
+  p | map2_;
+  p | is_identity_;
+}
+
+template <typename Map1, typename Map2>
+bool operator!=(const ProductOf2Maps<Map1, Map2>& lhs,
+                const ProductOf2Maps<Map1, Map2>& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+template <typename Map1, typename Map2, typename Map3>
+ProductOf3Maps<Map1, Map2, Map3>::ProductOf3Maps(Map1 map1, Map2 map2,
+                                                 Map3 map3) noexcept
+    : map1_(std::move(map1)),
+      map2_(std::move(map2)),
+      map3_(std::move(map3)),
+      is_identity_(map1_.is_identity() and map2_.is_identity() and
+                   map3_.is_identity()) {}
+
+template <typename Map1, typename Map2, typename Map3>
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, ProductOf3Maps<Map1, Map2, Map3>::dim>
+ProductOf3Maps<Map1, Map2, Map3>::operator()(
+    const std::array<T, dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const noexcept {
+  using UnwrappedT = tt::remove_cvref_wrap_t<T>;
+  return {
```

Comment:
I've thought about this and my conclusion was that it's not super easy to implement. Since so far it seems reasonable to restrict to 3 spatial dimensions we had decided to not try to implement a super generic product of N maps function.